src/core/Track.cpp

```diff
@@ -2653,15 +2653,29 @@ void Track::toggleSolo()
 			{
 				( *it )->m_mutedBeforeSolo = ( *it )->isMuted();
 			}
-			( *it )->setMuted( *it == this ? false : true );
+			// Don't mute AutomationTracks (keep their original state) unless we are on the sololegacybehavior mode
+			if( *it == this ){
+				( *it )->setMuted( false );
+			} else {
+				if( ConfigManager::inst()->value("app","sololegacybehavior").toInt() || ( *it )->type() != AutomationTrack ){
+					( *it )->setMuted( true );
+				}
+			}
 			if( *it != this )
 			{
 				( *it )->m_soloModel.setValue( false );
 			}
 		}
 		else if( !soloBefore )
 		{
-			( *it )->setMuted( ( *it )->m_mutedBeforeSolo );
+			// Unless we are on the sololegacybehavior mode, only restores the
+			// mute state if the track isn't an Automation Track
+			if( ConfigManager::inst()->value("app","sololegacybehavior").toInt() )
+			{
+				( *it )->setMuted( ( *it )->m_mutedBeforeSolo );
+			} else if( ( *it )->type() != AutomationTrack ) {
+				( *it )->setMuted( ( *it )->m_mutedBeforeSolo );
+			}
```

Comment:
It looks like both branches do the same thing, so I don't see why we couldn't merge them with an `||`.
```suggestion
			if( ConfigManager::inst()->value("app","sololegacybehavior").toInt()
				|| ( *it )->type() != AutomationTrack ) {
				( *it )->setMuted( ( *it )->m_mutedBeforeSolo );
			}
```

Reply:
True! Sometimes I think twice before nesting `if`s with `||`s for the sake of keeping the logic clearer, but in this case I think it's clear enough. I'll merge `Master`, make the change and push it!